src/win/fs.c

```diff
@@ -1201,7 +1200,7 @@ INLINE static int fs__stat_handle(HANDLE handle, uv_stat_t* statbuf,
 
   /* st_blocks contains the on-disk allocation size in 512-byte units. */
   statbuf->st_blocks =
-      file_info.StandardInformation.AllocationSize.QuadPart >> 9ULL;
+      file_info.StandardInformation.AllocationSize.QuadPart >> (uint64_t) 9;
```

Comment:
No cast necessary, I think? Plain 9 should suffice.

Reply:
I think this is one of the horrible edge cases of the promotion rules where an explicit cast is required (on at least one side of the binary operator) to preserve the exact meaning of the current expression.

Reply:
Can you walk me through that? QuadPart is 64 bits, isn't it? If it isn't, I'd expect the cast on the LHS, not the RHS.

Reply:
Ah, okay. I didn't realize QuadPart is signed. Can I suggest putting the cast on the LHS for legibility reasons? You can tell it wasn't completely obvious to me so it probably isn't obvious to others either.